src/explore-education-statistics-frontend/src/modules/table-tool/TableToolPage.tsx

```diff
@@ -43,7 +44,38 @@ const TableToolPage: NextPage<TableToolPageProps> = ({
   subjectMeta,
   themeMeta,
 }) => {
+  const router = useRouter();
   const [loadingFastTrack, setLoadingFastTrack] = useState(false);
+  const [updateActiveStep, setUpdateActiveStep] = useState<number | undefined>(
+    undefined,
+  );
+
+  useEffect(() => {
+    // Intercept the back button and activate the appropriate step
+    router.beforePopState(({ url }) => {
+      // going back to publication step
+      if (url === '/data-tables') {
+        setUpdateActiveStep(1);
+      }
+
+      // clicking back on any step after step 2 should take you to step 2
+      if (
+        url === '/data-tables/[publicationSlug]' ||
+        url === '/data-tables/[publicationSlug]/[releaseSlug]' ||
+        // when have gone directly to /data-tables/publication-slug the url has
+        // publicationSlug in the query as well so need to account for that.
+        url.startsWith('/data-tables/[publicationSlug]?publicationSlug=')
+      ) {
+        setUpdateActiveStep(2);
+      }
+
+      return true;
+    });
+
+    return () => {
+      router.beforePopState(() => true);
+    };
```

Comment:
Do we actually need this? This router API isn't like a typical event handler where we can unregister callbacks, so I'm not sure this is actually doing anything?

Reply:
You're right, it's not... I've removed it.